test/common/api_utils.py

```diff
@@ -0,0 +1,170 @@
+                    if not validate_obm or validate_obm_settings(client, nodeentry['id']):
+                        nodeIdList.append(nodeentry['id'])
+            else:
+                if 'sku' in nodeentry and skuid in nodeentry['sku']:
+                    if not validate_obm or validate_obm_settings(client, nodeentry['id']):
+                        nodeIdList.append(nodeentry['id'])
+        else:
+            # we are looking for a particular node with the requested OBM MAC
+            Api().nodes_get_obms_by_node_id(identifier=nodeentry['id'])
+            obmlistrsp = client.last_response
+            obmlist = loads(obmlistrsp.data)
+
+            # Determine if the list was obtained without error
+            if obmlistrsp.status != 200:
+                # There is no chance of getting the requested obm object using this node id, just 
+                # try another node.
+                print '**** Unable to retrieve obm list via API for node: {} status: ({}).\n'.format(nodeentry['id'],
+                                                                                                     obmlistrsp.status)
+                continue
+
+            # does this node use the requested OBM MAC
+            for obmentry in obmlist:
+                if obmentry['config'] and obmentry['config'].get('host') == obmmac:
+                    nodeIdList = [nodeentry['id']]
+                    break
+
+            # determine if the MAC was found, if so exit node loop
+            if nodeIdList:
+                break
+
+    # we now have a node ID list
+    if verbosity >= 3:
+        print "Node ID List:"
+        if nodeIdList:
+            print nodeIdList, '\n'
+        else:
+            print '**** Empty node ID list.\n'
+    return nodeIdList
+
+
+def validate_obm_settings(client, identifier):
+    """
+    The OBM objectis are obtained for the requested node. They are then searched to 
+    determine if at least one has OBM settings.
+    :param  client: reference to config.api2_0_config
+    :param  identifier: ID of node object
+    return: True  - node have OBM setting
+            False - node does not have OBM settings
+    """
+
+    Api().nodes_get_obms_by_node_id(identifier=identifier)
+    obmlistrsp = client.last_response
+    obmlist = loads(obmlistrsp.data)
+
+    # Determine if the list was obtained without error
+    if obmlistrsp.status != 200:
+        # There is no chance of getting the requested obm object using this node id, just 
+        # return False to indicate no OBM settings
+        print '**** Unable to retrieve obm list via API for node:: {} status: ({}).\n'.format(identifier,
+                                                                                              obmlistrsp.status)
+        return False
+
+    # does this node use the requested OBM MAC
+    for obmentry in obmlist:
+        if obmentry['config'] and obmentry['config'].get('user'):
```

Comment:
For some obm, like this one https://github.com/RackHD/on-http/blob/master/static/schemas/obms/panduit-obm-service.json,  there is no `user` field, but from your comment `determine if at least one has OBM settings`, this should return True as well. If so, you need a fix here.